File: Prong58b/MainForm.rb
require "mscorlib"
require "System.Windows.Forms, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089"
require "System.Drawing, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
def int(text)     return text.to_i end
def float(text)   return text.to_f end
def str(text)     return text.to_s end
def list(obj)     return obj.to_a  end
def len(arr)      return arr.length end
def input(msg="") print msg; return gets end
def print(*args)  Kernel.print(*args, "\n") end
def round(x, y)   return float((x * 10**y).round) / 10**y end
def range(*args)  if len(args) == 1 then 
    return  list((0...args[0]).step(1)); elsif len(args) == 2 then 
    return  list((args[0]...args[1]).step(1)); elsif len(args) == 3 then 
    return  list((args[0]...args[1]).step(args[2])) end; end
class MyRandom;   def randint(min, max) return rand(max-min) + min; end; 
    def random() return rand() end; 
    def shuffle(arr) return arr.shuffle end; 
    def choice(arr) return arr[randint(0, len(arr))] end; 
end; $random = MyRandom.new(); $math = Math
MessageBox = System::Windows::Forms::MessageBox
class MainForm < System::Windows::Forms::Form
	def initialize()
		self.InitializeComponent()
	end

	def InitializeComponent()
		@button1 = System::Windows::Forms::Button.new()
		@button2 = System::Windows::Forms::Button.new()
		@button3 = System::Windows::Forms::Button.new()
		@label1 = System::Windows::Forms::Label.new()
		@label2 = System::Windows::Forms::Label.new()
		@textBox1 = System::Windows::Forms::TextBox.new()
		@textBox3 = System::Windows::Forms::TextBox.new()
		@textBox5 = System::Windows::Forms::TextBox.new()
		self.SuspendLayout()
		# 
		# button1
		# 
		@button1.BackColor = System::Drawing::Color.Thistle
		@button1.Location = System::Drawing::Point.new(12, 299)
		@button1.Name = "button1"
		@button1.Size = System::Drawing::Size.new(132, 33)
		@button1.TabIndex = 0
		@button1.Text = "Calculate"
		@button1.UseVisualStyleBackColor = false
		# 
		# button2
		# 
		@button2.BackColor = System::Drawing::Color.Thistle
		@button2.Location = System::Drawing::Point.new(288, 299)
		@button2.Name = "button2"
		@button2.Size = System::Drawing::Size.new(132, 33)
		@button2.TabIndex = 1
		@button2.Text = "Exit"
		@button2.UseVisualStyleBackColor = false
		@button2.Click { |sender, e| self.Button2Click(sender, e) }
		# 
		# button3
		# 
		@button3.BackColor = System::Drawing::Color.Thistle
		@button3.Location = System::Drawing::Point.new(150, 299)
		@button3.Name = "button3"
		@button3.Size = System::Drawing::Size.new(132, 33)
		@button3.TabIndex = 2
		@button3.Text = "Clear"
		@button3.UseVisualStyleBackColor = false
		@button3.Click { |sender, e| self.Button3Click(sender, e) }
		# 
		# label1
		# 
		@label1.BackColor = System::Drawing::Color.LavenderBlush
		@label1.Location = System::Drawing::Point.new(12, 248)
		@label1.Name = "label1"
		@label1.Size = System::Drawing::Size.new(408, 23)
		@label1.TabIndex = 3
		# 
		# label2
		# 
		@label2.BackColor = System::Drawing::Color.LavenderBlush
		@label2.Location = System::Drawing::Point.new(12, 271)
		@label2.Name = "label2"
		@label2.Size = System::Drawing::Size.new(408, 23)
		@label2.TabIndex = 4
		@label2.Click { |sender, e| self.Label2Click(sender, e) }
		# 
		# textBox1
		# 
		@textBox1.Location = System::Drawing::Point.new(56, 112)
		@textBox1.Name = "textBox1"
		@textBox1.Size = System::Drawing::Size.new(100, 20)
		@textBox1.TabIndex = 5
		# 
		# textBox3
		# 
		@textBox3.Location = System::Drawing::Point.new(56, 164)
		@textBox3.Name = "textBox3"
		@textBox3.Size = System::Drawing::Size.new(100, 20)
		@textBox3.TabIndex = 7
		# 
		# textBox5
		# 
		@textBox5.Location = System::Drawing::Point.new(56, 138)
		@textBox5.Name = "textBox5"
		@textBox5.Size = System::Drawing::Size.new(100, 20)
		@textBox5.TabIndex = 9
		# 
		# MainForm
		# 
		self.BackColor = System::Drawing::Color.HotPink
		self.ClientSize = System::Drawing::Size.new(848, 344)
		self.Controls.Add(@textBox5)
		self.Controls.Add(@textBox3)
		self.Controls.Add(@textBox1)
		self.Controls.Add(@label2)
		self.Controls.Add(@label1)
		self.Controls.Add(@button3)
		self.Controls.Add(@button2)
		self.Controls.Add(@button1)
		self.Name = "MainForm"
		self.Text = "Prong58b"
		self.ResumeLayout(false)
		self.PerformLayout()
	end

	def Button3Click(sender, e)
		@label2.Text = ""
		@label1.Text = ""
		@textBox1.Text = ""
		@textBox5.Text = ""
		@textBox3.Text = ""
	end

	def Label2Click(sender, e)
		
	end

	def Button2Click(sender, e)
		Application.Exit
	end
end

File: Prog152b/MainForm.py
import System.Drawing
import System.Windows.Forms

from System.Drawing import *
from System.Windows.Forms import *

class MainForm(Form):
	def __init__(self):
		self.InitializeComponent()
	
	def InitializeComponent(self):
		self._textBox1 = System.Windows.Forms.TextBox()
		self._button1 = System.Windows.Forms.Button()
		self._button2 = System.Windows.Forms.Button()
		self._button3 = System.Windows.Forms.Button()
		self._listBox1 = System.Windows.Forms.ListBox()
		self.SuspendLayout()
		# 
		# textBox1
		# 
		self._textBox1.Location = System.Drawing.Point(150, 12)
		self._textBox1.Name = "textBox1"
		self._textBox1.Size = System.Drawing.Size(100, 20)
		self._textBox1.TabIndex = 0
		# 
		# button1
		# 
		self._button1.Location = System.Drawing.Point(69, 197)
		self._button1.Name = "button1"
		self._button1.Size = System.Drawing.Size(75, 23)
		self._button1.TabIndex = 2
		self._button1.Text = "Calculate"
		self._button1.UseVisualStyleBackColor = True
		self._button1.Click += self.Button1Click
		# 
		# button2
		# 
		self._button2.Location = System.Drawing.Point(150, 197)
		self._button2.Name = "button2"
		self._button2.Size = System.Drawing.Size(75, 23)
		self._button2.TabIndex = 3
		self._button2.Text = "Clear"
		self._button2.UseVisualStyleBackColor = True
		self._button2.Click += self.Button2Click
		# 
		# button3
		# 
		self._button3.Location = System.Drawing.Point(231, 197)
		self._button3.Name = "button3"
		self._button3.Size = System.Drawing.Size(75, 23)
		self._button3.TabIndex = 4
		self._button3.Text = "Exit"
		self._button3.UseVisualStyleBackColor = True
		self._button3.Click += self.Button3Click
		# 
		# listBox1
		# 
		self._listBox1.FormattingEnabled = True
		self._listBox1.Location = System.Drawing.Point(140, 52)
		self._listBox1.Name = "listBox1"
		self._listBox1.Size = System.Drawing.Size(120, 95)
		self._listBox1.TabIndex = 5
		self._listBox1.SelectedIndexChanged += self.ListBox1SelectedIndexChanged
		# 
		# MainForm
		# 
		self.ClientSize = System.Drawing.Size(423, 252)
		self.Controls.Add(self._listBox1)
		self.Controls.Add(self._button3)
		self.Controls.Add(self._button2)
		self.Controls.Add(self._button1)
		self.Controls.Add(self._textBox1)
		self.Name = "MainForm"
		self.Text = "Prog152b"
		self.ResumeLayout(False)
		self.PerformLayout()


	def ListBox1SelectedIndexChanged(self, sender, e):
		pass

	def Button1Click(self, sender, e):
		self._listBox1.Items.Clear()
		x = 0
		y = int(self._textBox1.Text)
		z = 0
		for num in range(0, y+1, +2):
			x += 2
			while x < y:
				x += 2
				self._listBox1.Items.Add(str(y) + "\t" + str(x))

	def Button2Click(self, sender, e):
		self._listBox1.Items.Clear()
		self._textBox1.Text = ""

	def Button3Click(self, sender, e):
		Application.Exit()